homeassistant/components/reolink/quality_scale.yaml

```diff
@@ -0,0 +1,71 @@
+rules:
+  # Bronze
+  action-setup:
+    status: done
+    comment: |
+      play_chime service is setup in async_setup
+      ptz_move service is setup in async_setup_entry since it is a entity_service
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+  # Silver
+  action-exceptions: done
+  config-entry-unloading: done
+  docs-configuration-parameters: done
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates:
+    status: done
+    comment: |
+      Coordinators are used and asyncio mutex locks ensure safe operation in the upstream lib
+      Parallel_update=0 set on all platforms
+  reauthentication-flow: done
+  test-coverage: done
+  # Gold
+  devices: done
+  diagnostics: done
+  discovery-update-info: done
+  discovery: done
+  docs-data-update: done
+  docs-examples: todo
+  docs-known-limitations: todo
+  docs-supported-devices: done
+  docs-supported-functions: done
+  docs-troubleshooting: done
+  docs-use-cases: todo
+  dynamic-devices: done
+  entity-category: done
+  entity-device-class: done
+  entity-disabled-by-default: done
+  entity-translations: done
+  exception-translations: done
+  icon-translations: done
+  reconfiguration-flow: done
+  repair-issues: done
+  stale-devices:
+    status: done
+    comment: |
+      For standalone cameras this does not apply: the integration should be removed.
+      For cameras connected to a NVR/Hub: the entities of a device are marked unavailable when power is unplugged. They can be removed using async_remove_config_entry_device.
+      Chimes can be uncoupled from the doorbell and removed from HA using async_remove_config_entry_device
+      Automatic removal lead to many user issues when a device was temporarily out of wifi range or disconnected from power, so not implemented anymore.
```

Comment:
How does the reolink app handle that?

Reply:
What do you mean?
About the standalone cameras, you mean?
For those just go to Settings->Devices&Services->Reolink->Overflow menu->Delete.
It has async_unload_entry implemented which will logout from the camera.

Reply:
The app will not automatically remove any setup camera. You always need to delete them manually, and in my opinion, we should not automatically delete any devices as Reolink offers not only POE cameras but also Wi-Fi and battery-based ones.
Especially battery ones can be offline when the battery is dead or during recharging. It would not be a great UX if HA would remove those devices in the meantime

Reply:
No but that is depending on how the API works. For example, if I disconnect my hue bulb, the API would still expose it, but with connected = false. In that case we know it's temporary. If the API would not expose that device when it's disconnected, we can't be certain if it's temporary, so hence the question

Reply:
Reolink does mark a device as beeing temporary offline, but if it keeps beeing offline for more than X hours (I believe something like 12 hours) or if the NVR is rebooted it will remove the device from the API.

I think the HomeHub does retain the devices better.

But in any case, the API was not trustworthy enough to start auto removing.

Reply:
Does it then also disappear from the app?

Reply:
So the app will still show it but the api doesn't expose it anymore?

Reply:
yes